stock_release_channel_shipment_advice/models/stock_release_channel.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockReleaseChannel(models.Model):
+    _inherit = "stock.release.channel"
+
+    shipment_planning_method = fields.Selection(
+        selection=[("none", "None"), ("simple", "Simple")], default="none"
+    )
+    picking_to_plan_ids = fields.Many2many(
+        comodel_name="stock.picking", compute="_compute_picking_to_plan_ids"
+    )
+    can_plan_shipment = fields.Boolean(compute="_compute_can_plan_shipment")
+    shipment_advice_ids = fields.One2many(
+        comodel_name="shipment.advice",
+        inverse_name="release_channel_id",
+        string="Shipment Advice",
+        readonly=True,
+    )
+
+    def button_show_shipment_advice(self):
+        self.ensure_one()
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "calendar,tree,form",
+            "res_model": self.shipment_advice_ids._name,
+            "domain": [("id", "in", self.shipment_advice_ids.ids)],
+            "context": self.env.context,
+        }
+
+    @api.depends("shipment_planning_method", "picking_to_plan_ids")
+    def _compute_can_plan_shipment(self):
+        for rec in self:
+            rec.can_plan_shipment = (
+                rec.picking_to_plan_ids
+                and rec.shipment_planning_method
+                and rec.shipment_planning_method != "none"
```

Comment:
```suggestion
                rec.shipment_planning_method
                and rec.shipment_planning_method != "none"
                and rec.picking_to_plan_ids
```

It's important to evaluate the cost-less expressions first. The evaluation stop on the first one returning False. That means that `picking_to_plan_ids` if rec.shipment_planning_method != "none"